src/Swarm/Language/Parse.hs

```diff
@@ -345,3 +348,22 @@ fully p = sc *> p <* eof
 --   a pretty-printed parse error message.
 readTerm :: Text -> Either Text Term
 readTerm = runParser (fully parseTerm)
+
+-- | A lower-level readTerm which returns the megaparsec bundle error
+--   for precise error reporting.
+readTerm' :: Text -> Either ParserError Term
+readTerm' = parse (runReaderT (fully parseTerm) DisallowAntiquoting) ""
+
+-- | A utility for converting a ParserError into a one line message:
+--   <line-nr>: <error-msg>
+showShortError :: ParserError -> String
```

Comment:
What's the purpose of showing a short one-line error instead of the whole pretty-printed parse error bundle? (I'm not implying it's bad, just trying to understand.)

Reply:
I thought this would help with non-lsp editor integration, like when running `M-x compile`, emacs automatically understand the `filename:line:message` format. Though now that we can have LSP, I think we should just use that for IDE.

Reply:
Instead of `--check`, i rename the feature `format`, where it currently prints the file when it is valid.